lib/ansible/modules/copy.py

```diff
@@ -654,7 +654,8 @@ def main():
                     module.fail_json(msg="Destination directory %s is not accessible" % (os.path.dirname(dest)))
             module.fail_json(msg="Destination directory %s does not exist" % (os.path.dirname(dest)))
 
-    if not os.access(os.path.dirname(b_dest), os.W_OK) and not module.params['unsafe_writes']:
+    # We only need write access to the parent when the file does not already exist (or we play it unsafe)
+    if not checksum_dest and not os.access(os.path.dirname(b_dest), os.W_OK) and not module.params['unsafe_writes']:
```

Comment:
we should already have handled missing file in previous branch, these changes don't seem to be needed either.